catalog/cataloger_merge_test.go

```diff
@@ -3,91 +3,136 @@ package catalog
 import (
 	"context"
 	"errors"
+	"fmt"
 	"strconv"
 	"testing"
 
 	"github.com/go-test/deep"
 
 	"github.com/davecgh/go-spew/spew"
+	"github.com/treeverse/lakefs/db"
 	"github.com/treeverse/lakefs/testutil"
 )
 
-func TestCataloger_Merge_FromParentNoChangesInChild(t *testing.T) {
-	ctx := context.Background()
-	c := testCataloger(t)
-	repository := testCatalogerRepo(t, ctx, c, "repo", "master")
-
-	// create 3 files on master and commit
-	for i := 0; i < 3; i++ {
-		testCatalogerCreateEntry(t, ctx, c, repository, "master", "/file"+strconv.Itoa(i), nil, "")
-	}
-	_, err := c.Commit(ctx, repository, "master", "commit to master", "tester", nil)
-	testutil.MustDo(t, "commit to master", err)
-
-	// create branch based on master
-	testCatalogerBranch(t, ctx, c, repository, "branch1", "master")
-
-	// add new file
-	const newFilename = "/file5"
-	testCatalogerCreateEntry(t, ctx, c, repository, "master", newFilename, nil, "")
-
-	// delete committed file
-	const delFilename = "/file1"
-	testutil.MustDo(t, "delete committed file on master",
-		c.DeleteEntry(ctx, repository, "master", delFilename))
-
-	// change/override committed file
-	const overFilename = "/file2"
-	testCatalogerCreateEntry(t, ctx, c, repository, "master", overFilename, nil, "seed1")
-
-	// commit, merge and verify
-	_, err = c.Commit(ctx, repository, "master", "second commit to master", "tester", nil)
-	testutil.MustDo(t, "second commit to master", err)
-
-	// before the merge - make sure we see the deleted file
-	_, err = c.GetEntry(ctx, repository, "branch1:HEAD", delFilename, GetEntryParams{})
-	if err != nil {
-		t.Fatalf("Get entry %s, expected to be found: %s", delFilename, err)
-	}
-
-	// merge master to branch1
-	res, err := c.Merge(ctx, repository, "master", "branch1", "tester", "", nil)
-	if err != nil {
-		t.Fatal("Merge from master to branch1 failed:", err)
-	}
-	if !IsValidReference(res.Reference) {
-		t.Fatalf("Merge reference = %s, expected valid reference", res.Reference)
-	}
-
-	testVerifyEntries(t, ctx, c, repository, "branch1", []testEntryInfo{
-		{Path: newFilename},
-		{Path: overFilename, Seed: "seed1"},
-		{Path: delFilename, Deleted: true},
+func validateMergeWithHooks(
```

Comment:
a bit too much, no? a simple test of any simple merge use case should verify that hooks works - looking for an issue on a failed test that is written like that will make it hard on the developer.